mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,466 @@
+        if self.num_eval_images > 0:
+            # Initialize data table
+            self._init_data_table()
+            # Add data to the table
+            self._add_ground_truth()
+            # Log ground truth data
+            self._log_data_table()
+
+        # Define a custom x-axes for validation metrics.
+        if self.log_eval_metrics:
+            self.wandb.define_metric('val/val_step')
+            self.wandb.define_metric('val/*', step_metric='val/val_step')
+
+    @master_only
+    def after_train_epoch(self, runner):
+        if self.log_eval_metrics:
+            if self.eval_hook.by_epoch:
+                if self.every_n_epochs(
+                        runner,
+                        self.eval_hook.interval) or self.is_last_epoch(runner):
+                    results = self.eval_hook.results
+                    eval_results = self.val_dataset.evaluate(
+                        results, logger='silent')
+                    for key, val in eval_results.items():
+                        if isinstance(val, str):
+                            continue
+                        self.wandb.log({f'val/{key}': val}, commit=False)
+                    self.wandb.log({'val/val_step': self.val_step})
+                    self.val_step += 1
+
+        if self.log_checkpoint:
+            if self.ckpt_hook.by_epoch:
+                if self.every_n_epochs(runner, self.ckpt_hook.interval) or (
+                        self.ckpt_hook.save_last
+                        and self.is_last_epoch(runner)):
+                    if self.log_checkpoint_metadata and self.eval_hook:
+                        metadata = self._get_ckpt_metadata(runner)
+                        aliases = [f'epoch_{runner.epoch+1}', 'latest']
+                        if self._is_best_ckpt(metadata):
+                            aliases.append('best')
+                        self._log_ckpt_as_artifact(self.ckpt_hook.out_dir,
+                                                   runner.epoch, aliases,
+                                                   metadata)
+                    else:
+                        aliases = [f'epoch_{runner.epoch+1}', 'latest']
+                        self._log_ckpt_as_artifact(self.ckpt_hook.out_dir,
+                                                   runner.epoch, aliases)
+
+        if self.num_eval_images > 0:
+            if self.eval_hook.by_epoch:
+                if self.every_n_epochs(
+                        runner,
+                        self.eval_hook.interval) or self.is_last_epoch(runner):
+                    results = self.eval_hook.results
+                    # Initialize evaluation table
+                    self._init_pred_table()
+                    # Log predictions
+                    self._log_predictions(results, runner.epoch + 1)
+                    # Log the table
+                    self._log_eval_table()
+
+    @master_only
+    def after_run(self, runner):
+        self.wandb.finish()
```

Comment:
As you can see I have overwritten the `after_run` method of `mmcv.WandbLoggerHook`. Do you want me to use `super()` instead?

The current `mmcv.WandbLoggerHook` logs the files with `output_prefix` as artifacts, however, this dedicated logger makes use of the artifacts better. If you think, it's best to not override, I will revert the change.